libs/isograph-react/src/core/read.ts

```diff
@@ -182,6 +201,7 @@ function readData<TReadFromStore>(
               networkRequest,
               networkRequestOptions,
               mutableEncounteredRecords,
+              field.concreteType,
```

Comment:
once we auto select `__typename`, we can do `field.concreteType ?? record.__typename`